packages/react-client/src/ReactFlightClient.js

```diff
@@ -902,6 +902,20 @@ function waitForReference<T>(
       handler.value = parentObject[key];
     }
 
+    // If the parent object is an unparsed React element tuple and its outlined
+    // props have now been resolved, we also need to update the props of the
+    // parsed element object (i.e. handler.value).
+    if (
+      parentObject[0] === REACT_ELEMENT_TYPE &&
+      key === '3' &&
```

Comment:
Is this the first time we're doing this kind of check? Maybe we should put these indices into variables to make it clearer what we're referencing here.

Reply:
There are at least two more cases:
https://github.com/facebook/react/blob/927ba5799a506e5f10ee71b3df8f59296550d29f/packages/react-client/src/ReactFlightClient.js#L901
https://github.com/facebook/react/blob/927ba5799a506e5f10ee71b3df8f59296550d29f/packages/react-client/src/ReactFlightClient.js#L1138
Also deferring the answer to @sebmarkbage.

Reply:
I think in this case, this problem is not limited to this particular slot though. It can happen to any key. Like the "type" and the "owner" (in DEV only) too.

So I think we probably should handle every key that an element can have and map it to its property name.